### PR TITLE
Fix dropout for recurrent dropout

### PR DESCRIPTION
**Pull request by sachinprasadhs**
Fix typo for the the `recurrent dropout` implementation to use `recurrent_dropout` instead of `dropout`.

recurrent_dropout has no effect in the below usage as per existing implementation.

`layer = keras.layers.SimpleRNN(10, recurrent_dropout=0.5)` 

Fixes: https://github.com/keras-team/keras/issues/19395

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
